bridges/otelzerolog/hook.go

```diff
@@ -0,0 +1,194 @@
+		c.provider = global.GetLoggerProvider()
+	}
+
+	return c
+}
+
+func (c config) logger(name string) log.Logger {
+	var opts []log.LoggerOption
+	if c.version != "" {
+		opts = append(opts, log.WithInstrumentationVersion(c.version))
+	}
+	if c.schemaURL != "" {
+		opts = append(opts, log.WithSchemaURL(c.schemaURL))
+	}
+	return c.provider.Logger(name, opts...)
+}
+
+// Option configures a SeverityHook.
+type Option interface {
+	apply(config) config
+}
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithVersion returns an [Option] that configures the version of the
+// [log.Logger] used by a [SeverityHook]. The version should be the version of the
+// package that is being logged.
+func WithVersion(version string) Option {
+	return optFunc(func(c config) config {
+		c.version = version
+		return c
+	})
+}
+
+// WithSchemaURL returns an [Option] that configures the semantic convention
+// schema URL of the [log.Logger] used by a [SeverityHook]. The schemaURL should be
+// the schema URL for the semantic conventions used in log records.
+func WithSchemaURL(schemaURL string) Option {
+	return optFunc(func(c config) config {
+		c.schemaURL = schemaURL
+		return c
+	})
+}
+
+// WithLoggerProvider returns an [Option] that configures [log.LoggerProvider]
+// used by a [SeverityHook].
+//
+// By default if this Option is not provided, the SeverityHook will use the global
+// LoggerProvider.
+func WithLoggerProvider(provider log.LoggerProvider) Option {
+	return optFunc(func(c config) config {
+		c.provider = provider
+		return c
+	})
+}
+
+// NewSeverityHook returns a new [SeverityHook] to be used as a [Zerolog.Hook].
+//
+// If [WithLoggerProvider] is not provided, the returned SeverityHook will use the
+// global LoggerProvider.
+func NewSeverityHook(name string, options ...Option) *SeverityHook {
+	cfg := newConfig(options)
+	return &SeverityHook{
+		logger: cfg.logger(name),
+	}
+}
+
+// // Hook is a [zerolog.Hook] that sends all logging records it receives to
+// OpenTelemetry. See package documentation for how conversions are made.
+type SeverityHook struct {
+	logger log.Logger
+	levels zerolog.Level
+}
+
+// Levels returns the list of log levels we want to be sent to OpenTelemetry.
+func (h *SeverityHook) Levels() zerolog.Level {
+	return h.levels
+}
+
+// Run handles the passed record, and sends it to OpenTelemetry.
+func (h SeverityHook) Run(e *zerolog.Event, level zerolog.Level, msg string) error {
+	if level != zerolog.NoLevel {
+		e.Str("severity", level.String())
+	}
+	h.logger.Emit(e.GetCtx(), h.convertEvent(e, level, msg))
+	return nil
+}
+
+func (h *SeverityHook) convertEvent(e *zerolog.Event, level zerolog.Level, msg string) log.Record {
+	var record log.Record
+	record.SetTimestamp(zerolog.TimestampFunc())
+	record.SetBody(log.StringValue(msg))
+	const sevOffset = zerolog.Level(log.SeverityDebug) - zerolog.DebugLevel
+	record.SetSeverity(log.Severity(level + sevOffset))
```

Comment:
This assumes the same scaling factor. OTel levels to not scale at this rate: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber